# Manual Checking of Rumah123 Scraped Data

Objective: To ensure if the data is scraped correctly

# Preface

Data of Rumah123 house listing of Balikpapan (city closest to Ibu Kota Nusantara) has been scraped. The next task is to manually cross-check with the website if the scraped data is accurate. 

To check, the author has been informed to just do sampling per 100 indexes. This sampling is not done randomly as the issue is thought to be in the later pages.

We'll be using this notebook to manually validate and record the results. Random number for every 100 index will be pulled using the `random` built-in module.

# Lib Loading


In [2]:
import random
import pandas as pd

# Data Loading

In [4]:
# load and see data
df = pd.read_csv('./datasets/rumah123_fix_data.csv')

df.head()

judul  \
0  RUMAH 2 LANTAI TENGAH KOTA PERUMAHAN BUKIT DAM...   
1  Rumah Luas Lokasi Strategis di Gunung Samarind...   
2  Rumah Minimalis Baru Renovasi Untuk Keluarga M...   
3  DIJUAL CEPAT RUMAH DI JALAN SWADAYA, KARANG RE...   
4       Rumah Minimalis Modern Smart Home Tanah Luas   

                                           hyperlink  \
0  https://www.rumah123.com//properti/balikpapan/...   
1  https://www.rumah123.com//properti/balikpapan/...   
2  https://www.rumah123.com//properti/balikpapan/...   
3  https://www.rumah123.com//properti/balikpapan/...   
4  https://www.rumah123.com//properti/balikpapan/...   

                                      img_hyperlinks  \
0  https://picture.rumah123.com/r123/720x420-crop...   
1  https://picture.rumah123.com/r123-images/720x4...   
2  https://picture.rumah123.com/r123-images/720x4...   
3  https://picture.rumah123.com/r123/720x420-crop...   
4  https://picture.rumah123.com/r123-images/720x4...   

                                               Judul          Harga  \
0  RUMAH 2 LANTAI TENGAH KOTA PERUMAHAN BUKIT DAM...  Rp 1,5 Miliar   
1  Rumah Luas Lokasi Strategis di Gunung Samarind...  Rp 3,1 Miliar   
2  Rumah Minimalis Baru Renovasi Untuk Keluarga M...    Rp 600 Juta   
3  DIJUAL CEPAT RUMAH DI JALAN SWADAYA, KARANG RE...    Rp 510 Juta   
4       Rumah Minimalis Modern Smart Home Tanah Luas    Rp 850 Juta   

                           Lokasi  Kamar Tidur  Kamar Mandi Luas Tanah  \
0  Balikpapan Selatan, Balikpapan          5.0          3.0     144 m²   
1    Balikpapan Utara, Balikpapan          3.0          2.0     534 m²   
2    Balikpapan Utara, Balikpapan          2.0          2.0      80 m²   
3   Balikpapan Tengah, Balikpapan          3.0          2.0     201 m²   
4    Balikpapan Utara, Balikpapan          3.0          3.0     120 m²   

  Luas Bangunan  ...  Ruang Tamu Kondisi Perabotan Jumlah Lantai  \
0        207 m²  ...       Tidak    Semi Furnished           2.0   
1        200 m²  ...          Ya               NaN           1.0   
2        140 m²  ...          Ya    Semi Furnished           1.0   
3        150 m²  ...       Tidak       Unfurnished           1.0   
4         65 m²  ...          Ya       Unfurnished           2.0   

       Pemandangan  Terjangkau Internet  Lebar Jalan     Sumber Air   Hook  \
0  Pemukiman Warga                   Ya      2 Mobil  PAM atau PDAM  Tidak   
1              NaN                   Ya          NaN  PAM atau PDAM  Tidak   
2              NaN                   Ya          NaN  PAM atau PDAM  Tidak   
3              NaN                Tidak          NaN            NaN  Tidak   
4  Pemukiman Warga                   Ya          NaN      Sumur Bor  Tidak   

  Kondisi Properti  Tipe Iklan  
0            Bagus      Dijual  
1            Bagus      Dijual  
2            Bagus      Dijual  
3            Bagus      Dijual  
4             Baru      Dijual  

[5 rows x 27 columns]

In [8]:
# see how many rows
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1467 entries, 0 to 1466
Data columns (total 27 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   judul                1467 non-null   object 
 1   hyperlink            1467 non-null   object 
 2   img_hyperlinks       1467 non-null   object 
 3   Judul                1453 non-null   object 
 4   Harga                1453 non-null   object 
 5   Lokasi               1453 non-null   object 
 6   Kamar Tidur          1403 non-null   float64
 7   Kamar Mandi          1402 non-null   float64
 8   Luas Tanah           1453 non-null   object 
 9   Luas Bangunan        1453 non-null   object 
 10  Carport              262 non-null    float64
 11  Tipe Properti        1453 non-null   object 
 12  Sertifikat           1452 non-null   object 
 13  Daya Listrik         1402 non-null   object 
 14  Kamar Pembantu       243 non-null    float64
 15  Dapur                232 non-null    f

Insight : there are 1467 rows with a lot of columns having empty rows, such as Carport, Dapur, Kondisi Perabotan, etc.

In [48]:
# code so that the hyperlinks are not cut off
pd.set_option('display.max_colwidth', None)

# Random Number Generator for Sampling

In [55]:
# made all a comment as to not change the random number

# # making random number generator every 100 index
# a = [0, 100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100, 1200, 1300, 1400]
# b= [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 1100, 1200, 1300, 1400, 1467]

# index_to_sample = []

# for i,n in enumerate(a):
#     random_number_loop = random.randint(n, b[i])
#     index_to_sample.append(random_number_loop)

# index_to_sample

Random numbers generated : 44, 172, 292, 371, 492, 589, 662, 784, 828, 941, 1089, 1121, 1278, 1334, 1461

# Manual Validation

In [51]:
df_1 = pd.DataFrame(df.iloc[44])

df_1

44
judul                                                                                                             RUMAH MURAH SIAP HUNI DEKAT BANDARA DAN KANTOR KEDINASAN
hyperlink                                                                                                       https://www.rumah123.com//properti/balikpapan/hos15901386/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho15/15901386/original/hos15901386-rumah-di-jual-di-balikpapan-selatan-balikpapan-1703831241.jpg
Judul                                                                                                             RUMAH MURAH SIAP HUNI DEKAT BANDARA DAN KANTOR KEDINASAN
Harga                                                                                                                                                          Rp 750 Juta
Lokasi                                                                                                                                      Balikpapan Selatan, Balikpapan
Kamar Tidur                                                                                                                                                            3.0
Kamar Mandi                                                                                                                                                            2.0
Luas Tanah                                                                                                                                                          150 m²
Luas Bangunan                                                                                                                                                       120 m²
Carport                                                                                                                                                                NaN
Tipe Properti                                                                                                                                                        Rumah
Sertifikat                                                                                                                                      SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                     1300 Watt
Kamar Pembantu                                                                                                                                                         NaN
Dapur                                                                                                                                                                  NaN
Ruang Makan                                                                                                                                                            NaN
Ruang Tamu                                                                                                                                                             NaN
Kondisi Perabotan                                                                                                                                                      NaN
Jumlah Lantai                                                                                                                                                          NaN
Pemandangan                                                                                                                                                            NaN
Terjangkau Internet                                                                                                                                                    NaN
Lebar Jalan                                                                                                                                                            NaN
Sumber Air                                                      

Insight : Index 44 same as webpage
---

In [52]:
df_1 = pd.DataFrame(df.iloc[172])

df_1

172
judul                                                                                                                            BALIKPAPAN RUMAH PUPUK TENGAH KOTA (LH52)
hyperlink                                                                                                       https://www.rumah123.com//properti/balikpapan/hos14514417/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho14/14514417/original/hos14514417-rumah-di-jual-di-balikpapan-selatan-balikpapan-1692333938.jpg
Judul                                                                                                                                           BALIKPAPAN PERUMAHAN BDS 2
Harga                                                                                                                                                          Rp 675 Juta
Lokasi                                                                                                                                      Balikpapan Selatan, Balikpapan
Kamar Tidur                                                                                                                                                            4.0
Kamar Mandi                                                                                                                                                            3.0
Luas Tanah                                                                                                                                                          250 m²
Luas Bangunan                                                                                                                                                       216 m²
Carport                                                                                                                                                                1.0
Tipe Properti                                                                                                                                                        Rumah
Sertifikat                                                                                                                                      SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                     3500 Watt
Kamar Pembantu                                                                                                                                                         1.0
Dapur                                                                                                                                                                  1.0
Ruang Makan                                                                                                                                                          Tidak
Ruang Tamu                                                                                                                                                           Tidak
Kondisi Perabotan                                                                                                                                              Unfurnished
Jumlah Lantai                                                                                                                                                          1.0
Pemandangan                                                                                                                                                            NaN
Terjangkau Internet                                                                                                                                                  Tidak
Lebar Jalan                                                                                                                                                        2 Mobil
Sumber Air                                                     

Insight : Index 172 has different price from the webpage, others same as webpage in URL.
---

- Webpage says "Rp 1,9 Miliar"? 
- 'judul' and 'Judul' columns are not the same.
- The features and URL are that of 'BALIKPAPAN RUMAH PUPUK TENGAH KOTA'

In [56]:
df_1 = pd.DataFrame(df.iloc[292])

df_1

292
judul                                                                                                               RUMAH TUMBUH MURAH DALAM PERUMAHAN ELITE BALIKPAPAN
hyperlink                                                                                                    https://www.rumah123.com//properti/balikpapan/hos10146593/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho10/10146593/original/hos10146593-rumah-di-jual-di-balikpapan-baru-balikpapan-1650935005.jpg
Judul                                                                                       RUMAH HOMY SIAP HUNI TIGA KAMAR TIDUR DALAM PERUMAHAN WIKA ELITE BALIKPAPAN
Harga                                                                                                                                                    Rp 1,42 Miliar
Lokasi                                                                                                                                        Gn. Samarinda, Balikpapan
Kamar Tidur                                                                                                                                                         2.0
Kamar Mandi                                                                                                                                                         1.0
Luas Tanah                                                                                                                                                       140 m²
Luas Bangunan                                                                                                                                                    100 m²
Carport                                                                                                                                                             NaN
Tipe Properti                                                                                                                                                     Rumah
Sertifikat                                                                                                                                   SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                  2200 Watt
Kamar Pembantu                                                                                                                                                      1.0
Dapur                                                                                                                                                               NaN
Ruang Makan                                                                                                                                                       Tidak
Ruang Tamu                                                                                                                                                        Tidak
Kondisi Perabotan                                                                                                                                                   NaN
Jumlah Lantai                                                                                                                                                       NaN
Pemandangan                                                                                                                                                         NaN
Terjangkau Internet                                                                                                                                               Tidak
Lebar Jalan                                                                                                                                                         NaN
Sumber Air                                                                                                                          

Insight: Index 292 has different price and location from the webpage, others same as webpage in URL.
---

- Location is Balikpapan Baru on the webpage
- Price said to be Rp 1,2 Miliar
- URL and features are of 'RUMAH TUMBUH MURAH DALAM PERUMAHAN ELITE BALIKPAPAN'
- 'judul' and 'Judul' columns are not the same

In [57]:
df_1 = pd.DataFrame(df.iloc[371])

df_1

371
judul                                                                                                         Rumah cantik minimalis, modern, lokasi ciputra hadap boulevard
hyperlink                                                                                                          https://www.rumah123.com//properti/balikpapan/hos4742484/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho47/4742484/original/hos4742484-rumah-di-jual-di-balikpapan-selatan-balikpapan-15629406018727.jpg
Judul                                                                                              Kost Baru + Rumah 2 lantai, 28 + 4 Kamar di Gunung Sari, Posisi Strategis
Harga                                                                                                                                                            Rp 3 Miliar
Lokasi                                                                                                                                         Balikpapan Tengah, Balikpapan
Kamar Tidur                                                                                                                                                              4.0
Kamar Mandi                                                                                                                                                              5.0
Luas Tanah                                                                                                                                                            300 m²
Luas Bangunan                                                                                                                                                         300 m²
Carport                                                                                                                                                                  NaN
Tipe Properti                                                                                                                                                          Rumah
Sertifikat                                                                                                                                        SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                             NaN
Kamar Pembantu                                                                                                                                                           NaN
Dapur                                                                                                                                                                    NaN
Ruang Makan                                                                                                                                                              NaN
Ruang Tamu                                                                                                                                                               NaN
Kondisi Perabotan                                                                                                                                                Unfurnished
Jumlah Lantai                                                                                                                                                            1.0
Pemandangan                                                                                                                                                              NaN
Terjangkau Internet                                                                                                                                                      NaN
Lebar Jalan                                                                                                                                                              NaN
Sumber Air       

Insight: Index 371 has different price and location from the webpage, others same as webpage in URL.
---

- Location on the webpage is Balikpapan Selatan
- Price said to be Rp 3,5 Miliar
- URL and features are of 'Rumah cantik minimalis, modern, lokasi ciputra hadap boulevard'
- 'judul' and 'Judul' columns are not the same


In [58]:
df_1 = pd.DataFrame(df.iloc[492])

df_1

492
judul                                                                                                             rumah di hook dekat terminal bus dan harga murah
hyperlink                                                                                               https://www.rumah123.com//properti/balikpapan/hos12285175/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho12/12285175/original/hos12285175-rumah-di-jual-di-batu-ampar-balikpapan-1672624315.jpg
Judul                                                                                                                Rumah Cantik terasa di Villa dan terasa mewah
Harga                                                                                                                                                  Rp 1 Miliar
Lokasi                                                                                                                                Balikpapan Utara, Balikpapan
Kamar Tidur                                                                                                                                                    2.0
Kamar Mandi                                                                                                                                                    1.0
Luas Tanah                                                                                                                                                   96 m²
Luas Bangunan                                                                                                                                                45 m²
Carport                                                                                                                                                        NaN
Tipe Properti                                                                                                                                                Rumah
Sertifikat                                                                                                                              SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                             1300 Watt
Kamar Pembantu                                                                                                                                                 NaN
Dapur                                                                                                                                                          NaN
Ruang Makan                                                                                                                                                  Tidak
Ruang Tamu                                                                                                                                                   Tidak
Kondisi Perabotan                                                                                                                                      Unfurnished
Jumlah Lantai                                                                                                                                                  NaN
Pemandangan                                                                                                                                             Pegunungan
Terjangkau Internet                                                                                                                                          Tidak
Lebar Jalan                                                                                                                                                1 Mobil
Sumber Air                                                                                                                                           PAM atau PDAM
Hook                                                                                

Insight: Index 492 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Batu Ampar on the Webpage
- Price said to be Rp 850 Juta
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'rumah di hook dekat terminal bus dan harga murah'

In [59]:
df_1 = pd.DataFrame(df.iloc[558])

df_1

558
judul                                                                                                              BALIKPAPAN,  PONDOK KARYA AGUNG BESAR (Y140)
hyperlink                                                                                             https://www.rumah123.com//properti/balikpapan/hos8509999/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho85/8509999/original/hos8509999-rumah-di-jual-di-sepinggan-balikpapan-1625102294.jpg
Judul                                                                                                                             Rumah Asri View Lapangan Golf
Harga                                                                                                                                            Rp 2,64 Miliar
Lokasi                                                                                                                           Balikpapan Selatan, Balikpapan
Kamar Tidur                                                                                                                                                 5.0
Kamar Mandi                                                                                                                                                 2.0
Luas Tanah                                                                                                                                               403 m²
Luas Bangunan                                                                                                                                            200 m²
Carport                                                                                                                                                     NaN
Tipe Properti                                                                                                                                             Rumah
Sertifikat                                                                                                                           SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                          2200 Watt
Kamar Pembantu                                                                                                                                              NaN
Dapur                                                                                                                                                       NaN
Ruang Makan                                                                                                                                                 NaN
Ruang Tamu                                                                                                                                                  NaN
Kondisi Perabotan                                                                                                                                           NaN
Jumlah Lantai                                                                                                                                               NaN
Pemandangan                                                                                                                                                 NaN
Terjangkau Internet                                                                                                                                         NaN
Lebar Jalan                                                                                                                                                 NaN
Sumber Air                                                                                                                                                  NaN
Hook                                                                                                                                                        

Insight: Index 558 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Sepinggan on the Webpage
- Price said to be Rp 2,2 Miliar
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'BALIKPAPAN, PONDOK KARYA AGUNG BESAR (Y140)'

In [60]:
df_1 = pd.DataFrame(df.iloc[662])

df_1

662
judul                                                                                                   INVESTASI RUMAH KOST  DI PINGGIR JALAN UTAMA PATIMURA BATU AMPAR
hyperlink                                                                                                     https://www.rumah123.com//properti/balikpapan/hos14554819/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho14/14554819/original/hos14554819-rumah-di-jual-di-balikpapan-utara-balikpapan-1692754367.jpg
Judul                                                                                                                                             BALIKPAPAN PERUM BDS 2
Harga                                                                                                                                                        Rp 700 Juta
Lokasi                                                                                                                                    Balikpapan Selatan, Balikpapan
Kamar Tidur                                                                                                                                                         29.0
Kamar Mandi                                                                                                                                                         29.0
Luas Tanah                                                                                                                                                        550 m²
Luas Bangunan                                                                                                                                                    1260 m²
Carport                                                                                                                                                              NaN
Tipe Properti                                                                                                                                                      Rumah
Sertifikat                                                                                                                                    SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                  13900 Watt
Kamar Pembantu                                                                                                                                                       NaN
Dapur                                                                                                                                                                NaN
Ruang Makan                                                                                                                                                          NaN
Ruang Tamu                                                                                                                                                           NaN
Kondisi Perabotan                                                                                                                                                    NaN
Jumlah Lantai                                                                                                                                                        NaN
Pemandangan                                                                                                                                                          NaN
Terjangkau Internet                                                                                                                                                  NaN
Lebar Jalan                                                                                                                                                          NaN
Sumber Air                                                                                                   

Insight: Index 662 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Balikpapan Utara on the Webpage
- Price said to be Rp 16 Miliar (I mean it's a whole tempat kost so makes sense??)
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'INVESTASI RUMAH KOST DI PINGGIR JALAN UTAMA PATIMURA BATU AMPAR'

In [61]:
df_1 = pd.DataFrame(df.iloc[784])

df_1

784
judul                                                                                                                         Di  Rumah Mewah Siap Huni Di Balikpapan
hyperlink                                                                                                  https://www.rumah123.com//properti/balikpapan/hos15928262/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho15/15928262/original/hos15928262-rumah-di-jual-di-gn.-samarinda-balikpapan-1704172345.jpg
Judul                                                                                                                         RUMAH BALIKPAPAN BDI PUSAT KOTA 5 KAMAR
Harga                                                                                                                                                   Rp 1,5 Miliar
Lokasi                                                                                                                                     Gunung Bahagia, Balikpapan
Kamar Tidur                                                                                                                                                       2.0
Kamar Mandi                                                                                                                                                       1.0
Luas Tanah                                                                                                                                                     208 m²
Luas Bangunan                                                                                                                                                   86 m²
Carport                                                                                                                                                           NaN
Tipe Properti                                                                                                                                                   Rumah
Sertifikat                                                                                                                                 SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                2200 Watt
Kamar Pembantu                                                                                                                                                    NaN
Dapur                                                                                                                                                             NaN
Ruang Makan                                                                                                                                                       NaN
Ruang Tamu                                                                                                                                                        NaN
Kondisi Perabotan                                                                                                                                                 NaN
Jumlah Lantai                                                                                                                                                     NaN
Pemandangan                                                                                                                                                       NaN
Terjangkau Internet                                                                                                                                               NaN
Lebar Jalan                                                                                                                                                       NaN
Sumber Air                                                                                                                                                        NaN
Hook        

Insight: Index 784 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Gn. Samarinda on the Webpage, Gn. Bahagia is location of "RUMAH BALIKPAPAN BDI PUSAT KOTA 5 KAMAR"
- Price said to be Rp 1,1 Miliar ,
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'Di Rumah Mewah Siap Huni Di Balikpapan'

In [62]:
df_1 = pd.DataFrame(df.iloc[828])

df_1

828
judul                                                                                                                       Rumah modern dekat Sekolah Islam dan Masjid
hyperlink                                                                                                    https://www.rumah123.com//properti/balikpapan/hos15145044/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho15/15145044/original/hos15145044-rumah-di-jual-di-balikpapan-baru-balikpapan-1697511182.jpg
Judul                                                                       RUMAH BARU READY STOCK & SIAP HUNI LOKASI ADA 2 TEMPAT PERUM PESONA ALAM & PERUM MANDASTANA
Harga                                                                                                                                                       Rp 415 Juta
Lokasi                                                                                                                                     Balikpapan Utara, Balikpapan
Kamar Tidur                                                                                                                                                         4.0
Kamar Mandi                                                                                                                                                         2.0
Luas Tanah                                                                                                                                                       200 m²
Luas Bangunan                                                                                                                                                    600 m²
Carport                                                                                                                                                             NaN
Tipe Properti                                                                                                                                                     Rumah
Sertifikat                                                                                                                                   SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                  5500 Watt
Kamar Pembantu                                                                                                                                                      NaN
Dapur                                                                                                                                                               NaN
Ruang Makan                                                                                                                                                       Tidak
Ruang Tamu                                                                                                                                                        Tidak
Kondisi Perabotan                                                                                                                                           Unfurnished
Jumlah Lantai                                                                                                                                                       NaN
Pemandangan                                                                                                                                             Pemukiman Warga
Terjangkau Internet                                                                                                                                               Tidak
Lebar Jalan                                                                                                                                                     2 Mobil
Sumber Air                                                                                                                          

Insight: Index 828 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Balikpapan Baru on the Webpage, Balikpapan Utara is location of "RUMAH BARU READY STOCK & SIAP HUNI LOKASI ADA 2 TEMPAT PERUM PESONA ALAM & PERUM MANDASTANA"
- Price on webpage is Rp 5 Miliar, 415 Juta is of 'RUMAH BARU READY STOCK & SIAP HUNI LOKASI ADA 2 TEMPAT PERUM PESONA ALAM & PERUM MANDASTANA' listing
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'Rumah modern dekat Sekolah Islam dan Masjid'

In [63]:
df_1 = pd.DataFrame(df.iloc[941])

df_1

941
judul                                                                                                                            RUMAH MINIMALIS DI PERUMAHAN INDRA KILA
hyperlink                                                                                                     https://www.rumah123.com//properti/balikpapan/hos13249706/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho13/13249706/original/hos13249706-rumah-di-jual-di-balikpapan-utara-balikpapan-1679984357.jpg
Judul                                                                                                                                   Rumah di jalan Mulawarman Lamaru
Harga                                                                                                                                                        Rp 400 Juta
Lokasi                                                                                                                                      Balikpapan Timur, Balikpapan
Kamar Tidur                                                                                                                                                          4.0
Kamar Mandi                                                                                                                                                          3.0
Luas Tanah                                                                                                                                                         91 m²
Luas Bangunan                                                                                                                                                     180 m²
Carport                                                                                                                                                              NaN
Tipe Properti                                                                                                                                                      Rumah
Sertifikat                                                                                                                                       HGB - Hak Guna Bangunan
Daya Listrik                                                                                                                                                   4400 Watt
Kamar Pembantu                                                                                                                                                       NaN
Dapur                                                                                                                                                                NaN
Ruang Makan                                                                                                                                                          NaN
Ruang Tamu                                                                                                                                                           NaN
Kondisi Perabotan                                                                                                                                                    NaN
Jumlah Lantai                                                                                                                                                        NaN
Pemandangan                                                                                                                                                          NaN
Terjangkau Internet                                                                                                                                                  NaN
Lebar Jalan                                                                                                                                                          NaN
Sumber Air                                                                                                   

Insight: Index 941 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Balikpapan Utara on the Webpage
- Price is Rp 1,8 Miliar on the webpage
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'RUMAH MINIMALIS DI PERUMAHAN INDRA KILA'

In [64]:
df_1 = pd.DataFrame(df.iloc[1089])

df_1

1089
judul                                                                                                      Hunian baru konsep Villa ditengah kota balikpapan 
hyperlink                                                                                          https://www.rumah123.com//properti/balikpapan/hos15048371/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho15/15048371/original/hos15048371-rumah-di-jual-di-damai-balikpapan-1696819627.jpg
Judul                                                                                                                BALIKPAPAN, RUMAH LUAS DI MANGGAR (Y515)
Harga                                                                                                                                          Rp 1,35 Miliar
Lokasi                                                                                                                                    Batakan, Balikpapan
Kamar Tidur                                                                                                                                               3.0
Kamar Mandi                                                                                                                                               3.0
Luas Tanah                                                                                                                                             125 m²
Luas Bangunan                                                                                                                                          135 m²
Carport                                                                                                                                                   2.0
Tipe Properti                                                                                                                                           Rumah
Sertifikat                                                                                                                         SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                        2200 Watt
Kamar Pembantu                                                                                                                                            NaN
Dapur                                                                                                                                                     NaN
Ruang Makan                                                                                                                                             Tidak
Ruang Tamu                                                                                                                                              Tidak
Kondisi Perabotan                                                                                                                                 Unfurnished
Jumlah Lantai                                                                                                                                             NaN
Pemandangan                                                                                                                                   Pemukiman Warga
Terjangkau Internet                                                                                                                                     Tidak
Lebar Jalan                                                                                                                                           2 Mobil
Sumber Air                                                                                                                                          Sumur Bor
Hook                                                                                                                                                    Tidak
Kondisi Properti                             

Insight: Index 1089 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Damai, Balikpapan on the Webpage. Batakan, Balikpapan is of "BALIKPAPAN, RUMAH LUAS DI MANGGAR (Y515)' listing
- Price is Rp 850 Juta on the webpage, Rp 1.35 Miliar is of 'BALIKPAPAN, RUMAH LUAS DI MANGGAR (Y515)'
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'Hunian baru konsep Villa ditengah kota balikpapan'

In [65]:
df_1 = pd.DataFrame(df.iloc[1121])

df_1

1121
judul                                                                                                             rumah plus toko dekat bandara international dan rekreasi
hyperlink                                                                                                       https://www.rumah123.com//properti/balikpapan/hos14676411/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho14/14676411/original/hos14676411-rumah-di-jual-di-balikpapan-selatan-balikpapan-1693809469.jpg
Judul                                                                                                              RUMAH TIGA KAMAR TIDUR DALAM PERUMAHAN RAMAI BALIKPAPAN
Harga                                                                                                                                                        Rp 1,2 Miliar
Lokasi                                                                                                                                        Balikpapan Utara, Balikpapan
Kamar Tidur                                                                                                                                                            4.0
Kamar Mandi                                                                                                                                                            2.0
Luas Tanah                                                                                                                                                          330 m²
Luas Bangunan                                                                                                                                                       270 m²
Carport                                                                                                                                                                2.0
Tipe Properti                                                                                                                                                        Rumah
Sertifikat                                                                                                                                      SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                     3500 Watt
Kamar Pembantu                                                                                                                                                         NaN
Dapur                                                                                                                                                                  NaN
Ruang Makan                                                                                                                                                          Tidak
Ruang Tamu                                                                                                                                                           Tidak
Kondisi Perabotan                                                                                                                                              Unfurnished
Jumlah Lantai                                                                                                                                                          2.0
Pemandangan                                                                                                                                                Pemukiman Warga
Terjangkau Internet                                                                                                                                                  Tidak
Lebar Jalan                                                                                                                                                        3 Mobil
Sumber Air                                                    

Insight: Index 1121 has different price and location from the webpage, others same as webpage in URL.
---
- Location is Balikpapan Selatan, Balikpapan on the Webpage. 
- Price is Rp 950 Juta on the webpage
- 'judul' and 'Judul' columns are not the same
- URL and features are of 'rumah plus toko dekat bandara international dan rekreasi'

In [66]:
df_1 = pd.DataFrame(df.iloc[1278])

df_1

1278
judul                                                                                                           BALIKPAPAN RUMAH MEWAH KOMPLEKS PERUMAHAN BDI
hyperlink                                                                                          https://www.rumah123.com//properti/balikpapan/hos14805138/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho14/14805138/original/hos14805138-rumah-di-jual-di-damai-balikpapan-1694916226.jpg
Judul                                                                                                                    Rumah Minimalis Siap Huni Free Biaya
Harga                                                                                                                                             Rp 360 Juta
Lokasi                                                                                                                           Balikpapan Utara, Balikpapan
Kamar Tidur                                                                                                                                               5.0
Kamar Mandi                                                                                                                                               5.0
Luas Tanah                                                                                                                                             280 m²
Luas Bangunan                                                                                                                                          226 m²
Carport                                                                                                                                                   NaN
Tipe Properti                                                                                                                                           Rumah
Sertifikat                                                                                                                         SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                        4400 Watt
Kamar Pembantu                                                                                                                                            1.0
Dapur                                                                                                                                                     NaN
Ruang Makan                                                                                                                                               NaN
Ruang Tamu                                                                                                                                                NaN
Kondisi Perabotan                                                                                                                                         NaN
Jumlah Lantai                                                                                                                                             NaN
Pemandangan                                                                                                                                               NaN
Terjangkau Internet                                                                                                                                       NaN
Lebar Jalan                                                                                                                                               NaN
Sumber Air                                                                                                                                                NaN
Hook                                                                                                                                                      NaN
Kondisi Properti                             

Insight: Index 1278 has different price and location from the webpage, others same as webpage in URL.
---

In [67]:
df_1 = pd.DataFrame(df.iloc[1334])

df_1

1334
judul                                                                                  FULLY FURNISHED RUMAH  CONTOH *NEW LYCHEE* 2 LANTAI DAUN VILLAGE BALIKPAPAN KAL-TIM
hyperlink                                                                                                       https://www.rumah123.com//properti/balikpapan/hos15652084/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho15/15652084/original/hos15652084-rumah-di-jual-di-balikpapan-selatan-balikpapan-1701476428.jpg
Judul                                                                                           RUMAH SIAP HUNI SATU LANTAI DENGAN DESAIN MINIMALIS LOKASI AMAN DAN NYAMAN
Harga                                                                                                                                                        Rp 1,5 Miliar
Lokasi                                                                                                                                        Balikpapan Utara, Balikpapan
Kamar Tidur                                                                                                                                                            3.0
Kamar Mandi                                                                                                                                                            2.0
Luas Tanah                                                                                                                                                          240 m²
Luas Bangunan                                                                                                                                                       116 m²
Carport                                                                                                                                                                NaN
Tipe Properti                                                                                                                                                        Rumah
Sertifikat                                                                                                                                         HGB - Hak Guna Bangunan
Daya Listrik                                                                                                                                                     1300 Watt
Kamar Pembantu                                                                                                                                                         1.0
Dapur                                                                                                                                                                  1.0
Ruang Makan                                                                                                                                                          Tidak
Ruang Tamu                                                                                                                                                           Tidak
Kondisi Perabotan                                                                                                                                                      NaN
Jumlah Lantai                                                                                                                                                          2.0
Pemandangan                                                                                                                                                Pemukiman Warga
Terjangkau Internet                                                                                                                                                     Ya
Lebar Jalan                                                                                                                                                        2 Mobil
Sumber Air                                                    

Insight: Index 1334 has different price and location from the webpage, others same as webpage in URL.
---

In [68]:
df_1 = pd.DataFrame(df.iloc[1461])

df_1

1461
judul                                                                                                                 RUMAH LUAS TERAWAT DALAM PERUMAHAN DI BALIKPAPAN
hyperlink                                                                                                    https://www.rumah123.com//properti/balikpapan/hos8827526/
img_hyperlinks       https://picture.rumah123.com/r123/720x420-crop/house/ho88/8827526/original/hos8827526-rumah-di-jual-di-balikpapan-utara-balikpapan-1631327326.jpg
Judul                                                                                                                                                              NaN
Harga                                                                                                                                                              NaN
Lokasi                                                                                                                                                             NaN
Kamar Tidur                                                                                                                                                        4.0
Kamar Mandi                                                                                                                                                        2.0
Luas Tanah                                                                                                                                                      215 m²
Luas Bangunan                                                                                                                                                   300 m²
Carport                                                                                                                                                            NaN
Tipe Properti                                                                                                                                                    Rumah
Sertifikat                                                                                                                                  SHM - Sertifikat Hak Milik
Daya Listrik                                                                                                                                                 3500 Watt
Kamar Pembantu                                                                                                                                                     1.0
Dapur                                                                                                                                                              NaN
Ruang Makan                                                                                                                                                        NaN
Ruang Tamu                                                                                                                                                         NaN
Kondisi Perabotan                                                                                                                                                  NaN
Jumlah Lantai                                                                                                                                                      NaN
Pemandangan                                                                                                                                                        NaN
Terjangkau Internet                                                                                                                                                NaN
Lebar Jalan                                                                                                                                                        NaN
Sumber Air                                                                                                                                                

Insight: Index 1461 has different price and location from the webpage, others same as webpage in URL.
---

# Conclusion

There is discrepancy between price, location, and the two 'judul' and 'Judul' column where:
- House features data and URL matches 'judul'
- Price and Location matches 'Judul'

# Solution: Inner Join the two separate csv on 'Judul'

So the problem lies with how the scraped data and scraped hyperlinks are concatenated. There's different title bcs both of those tables have the field 'judul' and the concatenation was not on the same judul, but instead a basic concatenation.

judul from the hyperlink table will be used as inner join base

In [69]:
# load the csvs to join

csv_hyperlink = pd.read_csv('./datasets/rumah123_hyperlink.csv')
csv_data = pd.read_csv('./datasets/rumah123_data_single_raw.csv')

In [70]:
# look at column names, especially 'Judul'

csv_data.head()

Judul  \
0  RUMAH 2 LANTAI TENGAH KOTA PERUMAHAN BUKIT DAMAI INDAH BALIKPAPAN KAL-TIM   
1     Rumah Luas Lokasi Strategis di Gunung Samarinda Baru, Balikpapan Utara   
2                      Rumah Minimalis Baru Renovasi Untuk Keluarga Muslimin   
3                DIJUAL CEPAT RUMAH DI JALAN SWADAYA, KARANG REJO BALIKPAPAN   
4                               Rumah Minimalis Modern Smart Home Tanah Luas   

           Harga                          Lokasi  Kamar Tidur  Kamar Mandi  \
0  Rp 1,5 Miliar  Balikpapan Selatan, Balikpapan          5.0          3.0   
1  Rp 3,1 Miliar    Balikpapan Utara, Balikpapan          3.0          2.0   
2    Rp 600 Juta    Balikpapan Utara, Balikpapan          2.0          2.0   
3    Rp 510 Juta   Balikpapan Tengah, Balikpapan          3.0          2.0   
4    Rp 850 Juta    Balikpapan Utara, Balikpapan          3.0          3.0   

  Luas Tanah Luas Bangunan  Carport Tipe Properti                  Sertifikat  \
0     144 m²        207 m²      1.0         Rumah  SHM - Sertifikat Hak Milik   
1     534 m²        200 m²      3.0         Rumah  SHM - Sertifikat Hak Milik   
2      80 m²        140 m²      1.0         Rumah  SHM - Sertifikat Hak Milik   
3     201 m²        150 m²      NaN         Rumah  SHM - Sertifikat Hak Milik   
4     120 m²         65 m²      1.0         Rumah  SHM - Sertifikat Hak Milik   

   ... Ruang Tamu  Kondisi Perabotan  Jumlah Lantai      Pemandangan  \
0  ...      Tidak     Semi Furnished            2.0  Pemukiman Warga   
1  ...         Ya                NaN            1.0              NaN   
2  ...         Ya     Semi Furnished            1.0              NaN   
3  ...      Tidak        Unfurnished            1.0              NaN   
4  ...         Ya        Unfurnished            2.0  Pemukiman Warga   

  Terjangkau Internet Lebar Jalan     Sumber Air   Hook Kondisi Properti  \
0                  Ya     2 Mobil  PAM atau PDAM  Tidak            Bagus   
1                  Ya         NaN  PAM atau PDAM  Tidak            Bagus   
2                  Ya         NaN  PAM atau PDAM  Tidak            Bagus   
3               Tidak         NaN            NaN  Tidak            Bagus   
4                  Ya         NaN      Sumur Bor  Tidak             Baru   

  Tipe Iklan  
0     Dijual  
1     Dijual  
2     Dijual  
3     Dijual  
4     Dijual  

[5 rows x 24 columns]

In [73]:
# look at column names, especially 'judul'

csv_hyperlink.head()

judul  \
0  RUMAH 2 LANTAI TENGAH KOTA PERUMAHAN BUKIT DAMAI INDAH BALIKPAPAN KAL-TIM   
1     Rumah Luas Lokasi Strategis di Gunung Samarinda Baru, Balikpapan Utara   
2                      Rumah Minimalis Baru Renovasi Untuk Keluarga Muslimin   
3                DIJUAL CEPAT RUMAH DI JALAN SWADAYA, KARANG REJO BALIKPAPAN   
4                               Rumah Minimalis Modern Smart Home Tanah Luas   

                                                    hyperlink  \
0  https://www.rumah123.com//properti/balikpapan/hos15945789/   
1  https://www.rumah123.com//properti/balikpapan/hos16239750/   
2  https://www.rumah123.com//properti/balikpapan/hos15643485/   
3  https://www.rumah123.com//properti/balikpapan/hos14193967/   
4  https://www.rumah123.com//properti/balikpapan/hos15998364/   

                                                                                                                                          img_hyperlinks  
0  https://picture.rumah123.com/r123/720x420-crop/house/ho15/15945789/original/hos15945789-rumah-di-jual-di-balikpapan-selatan-balikpapan-1704340035.jpg  
1                    https://picture.rumah123.com/r123-images/720x420-crop/customer/1964420/2024-02-01-02-35-27-030b4254-675e-46cc-b965-e52a0bb80d6d.jpg  
2                    https://picture.rumah123.com/r123-images/720x420-crop/customer/1926217/2023-12-01-06-10-42-94a69e3c-8b2e-4af7-9489-e15c0da42060.jpg  
3   https://picture.rumah123.com/r123/720x420-crop/house/ho14/14193967/original/hos14193967-rumah-di-jual-di-balikpapan-tengah-balikpapan-1689042171.jpg  
4                    https://picture.rumah123.com/r123-images/720x420-crop/customer/1355633/2024-01-23-07-46-31-33b01afa-3599-49a3-8e3f-2eca551661c0.png

Insight : notice how in csv_hyperlink it's 'judul' while on csv_data it's 'Judul' with a capital J. If we want to join these two tables on 'judul', the name would have to be the same for the program to recognize them as same.

In [72]:
# rename column 'Judul' in csv_data to 'judul'

csv_data.rename(columns={'Judul':"judul"}, inplace=True)

csv_data.head()

judul  \
0  RUMAH 2 LANTAI TENGAH KOTA PERUMAHAN BUKIT DAMAI INDAH BALIKPAPAN KAL-TIM   
1     Rumah Luas Lokasi Strategis di Gunung Samarinda Baru, Balikpapan Utara   
2                      Rumah Minimalis Baru Renovasi Untuk Keluarga Muslimin   
3                DIJUAL CEPAT RUMAH DI JALAN SWADAYA, KARANG REJO BALIKPAPAN   
4                               Rumah Minimalis Modern Smart Home Tanah Luas   

           Harga                          Lokasi  Kamar Tidur  Kamar Mandi  \
0  Rp 1,5 Miliar  Balikpapan Selatan, Balikpapan          5.0          3.0   
1  Rp 3,1 Miliar    Balikpapan Utara, Balikpapan          3.0          2.0   
2    Rp 600 Juta    Balikpapan Utara, Balikpapan          2.0          2.0   
3    Rp 510 Juta   Balikpapan Tengah, Balikpapan          3.0          2.0   
4    Rp 850 Juta    Balikpapan Utara, Balikpapan          3.0          3.0   

  Luas Tanah Luas Bangunan  Carport Tipe Properti                  Sertifikat  \
0     144 m²        207 m²      1.0         Rumah  SHM - Sertifikat Hak Milik   
1     534 m²        200 m²      3.0         Rumah  SHM - Sertifikat Hak Milik   
2      80 m²        140 m²      1.0         Rumah  SHM - Sertifikat Hak Milik   
3     201 m²        150 m²      NaN         Rumah  SHM - Sertifikat Hak Milik   
4     120 m²         65 m²      1.0         Rumah  SHM - Sertifikat Hak Milik   

   ... Ruang Tamu  Kondisi Perabotan  Jumlah Lantai      Pemandangan  \
0  ...      Tidak     Semi Furnished            2.0  Pemukiman Warga   
1  ...         Ya                NaN            1.0              NaN   
2  ...         Ya     Semi Furnished            1.0              NaN   
3  ...      Tidak        Unfurnished            1.0              NaN   
4  ...         Ya        Unfurnished            2.0  Pemukiman Warga   

  Terjangkau Internet Lebar Jalan     Sumber Air   Hook Kondisi Properti  \
0                  Ya     2 Mobil  PAM atau PDAM  Tidak            Bagus   
1                  Ya         NaN  PAM atau PDAM  Tidak            Bagus   
2                  Ya         NaN  PAM atau PDAM  Tidak            Bagus   
3               Tidak         NaN            NaN  Tidak            Bagus   
4                  Ya         NaN      Sumur Bor  Tidak             Baru   

  Tipe Iklan  
0     Dijual  
1     Dijual  
2     Dijual  
3     Dijual  
4     Dijual  

[5 rows x 24 columns]

In [106]:
csv_inner_join = csv_hyperlink.merge(csv_data, on='judul', how='inner')

csv_inner_join.tail()



judul  \
1793  Rumah Murah Harga Nego Dalam Perumahan   
1794             Perumahan Elit Pupuk Kaltim   
1795       Rumah Baru Minimalis Kpr Tanpa Dp   
1796                  BALIKPAPAN POKA (Y140)   
1797                 Rumah cantik tanah luas   

                                                      hyperlink  \
1793  https://www.rumah123.com//properti/balikpapan/hos8685808/   
1794  https://www.rumah123.com//properti/balikpapan/hos8219826/   
1795  https://www.rumah123.com//properti/balikpapan/hos8186149/   
1796  https://www.rumah123.com//properti/balikpapan/hos7941758/   
1797  https://www.rumah123.com//properti/balikpapan/hos7790080/   

                                                                                                                                               img_hyperlinks  \
1793    https://picture.rumah123.com/r123/720x420-crop/house/ho86/8685808/original/hos8685808-rumah-di-jual-di-balikpapan-timur-balikpapan-16287647172791.jpg   
1794      https://picture.rumah123.com/r123/720x420-crop/house/ho82/8219826/original/hos8219826-rumah-di-jual-di-balikpapan-selatan-balikpapan-1703827364.jpg   
1795    https://picture.rumah123.com/r123/720x420-crop/house/ho81/8186149/original/hos8186149-rumah-di-jual-di-balikpapan-utara-balikpapan-16284098149928.jpg   
1796               https://picture.rumah123.com/r123/720x420-crop/house/ho79/7941758/original/hos7941758-rumah-di-jual-di-sepinggan-balikpapan-1613803545.jpg   
1797  https://picture.rumah123.com/r123/720x420-crop/house/ho77/7790080/original/hos7790080-rumah-di-jual-di-balikpapan-selatan-balikpapan-16107687459208.jpg   

              Harga                          Lokasi  Kamar Tidur  Kamar Mandi  \
1793    Rp 150 Juta    Balikpapan Timur, Balikpapan          4.0          2.0   
1794  Rp 2,7 Miliar  Balikpapan Selatan, Balikpapan          2.0          1.0   
1795    Rp 395 Juta    Balikpapan Utara, Balikpapan          3.0          2.0   
1796  Rp 2,1 Miliar           Sepinggan, Balikpapan          4.0          2.0   
1797  Rp 2,5 Miliar  Balikpapan Selatan, Balikpapan          4.0          2.0   

     Luas Tanah Luas Bangunan  Carport  ... Ruang Tamu Kondisi Perabotan  \
1793     210 m²        120 m²      1.0  ...        NaN       Unfurnished   
1794     113 m²         36 m²      1.0  ...        NaN    Semi Furnished   
1795     162 m²         72 m²      1.0  ...        NaN         Furnished   
1796     356 m²        199 m²      NaN  ...        NaN               NaN   
1797     163 m²        100 m²      NaN  ...      Tidak               NaN   

     Jumlah Lantai  Pemandangan  Terjangkau Internet Lebar Jalan Sumber Air  \
1793           1.0          NaN                  NaN         NaN        NaN   
1794           1.0          NaN                  NaN         NaN        NaN   
1795           2.0          NaN                  NaN         NaN        NaN   
1796           NaN          NaN                  NaN         NaN        NaN   
1797           NaN          NaN                Tidak         NaN        NaN   

       Hook  Kondisi Properti Tipe Iklan  
1793    NaN    Butuh Renovasi     Dijual  
1794    NaN               NaN     Dijual  
1795    NaN               NaN     Dijual  
1796    NaN               NaN     Dijual  
1797  Tidak               NaN     Dijual  

[5 rows x 26 columns]

In [91]:
csv_inner_join.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1814 entries, 0 to 1813
Data columns (total 26 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   judul                1814 non-null   object 
 1   hyperlink            1814 non-null   object 
 2   img_hyperlinks       1814 non-null   object 
 3   Harga                1814 non-null   object 
 4   Lokasi               1814 non-null   object 
 5   Kamar Tidur          1737 non-null   float64
 6   Kamar Mandi          1736 non-null   float64
 7   Luas Tanah           1798 non-null   object 
 8   Luas Bangunan        1798 non-null   object 
 9   Carport              323 non-null    float64
 10  Tipe Properti        1798 non-null   object 
 11  Sertifikat           1797 non-null   object 
 12  Daya Listrik         1731 non-null   object 
 13  Kamar Pembantu       282 non-null    float64
 14  Dapur                283 non-null    float64
 15  Ruang Makan          888 non-null    o

The author will be changing the iloc number right below this markdown to the random numbers to more easily cross-check and write down conclusion.

In [109]:
csv_inner_join.iloc[44]

judul                                              TURUN HARGA!!!  CEPAT RUMAH CANTIK BARU SELESAI RENOVASI DI PERUMAHAN TAMANSARI BUKIT MUTIARA (WIKA) BALIKPAPAN KAL-TIM
hyperlink                                                                                                       https://www.rumah123.com//properti/balikpapan/hos16104213/
img_hyperlinks         https://picture.rumah123.com/r123/720x420-crop/house/ho16/16104213/original/hos16104213-rumah-di-jual-di-balikpapan-utara-balikpapan-1705654413.jpg
Harga                                                                                                                                                          Rp 600 Juta
Lokasi                                                                                                                                        Balikpapan Utara, Balikpapan
Kamar Tidur                                                                                                                                      

Insight : Validation status:
- Index number 44: same as website
- Index number 172
- Index number 292
- Index number 371
- Index number 492
- Index number 558
- Index number 662
- Index number 784
- Index number 828
- Index number 941
- Index number 1089
- Index number 1121
- Index number 1278
- Index number 1334
- Index number 1461: Different